### ISSUE TITLE
Mono export templates lack mono shared libs on Linux system using /usr/lib64

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#32558]
**Godot version:**
3.2 master (5a23ab61)

**OS/device including version:**
Linux, RPM-based distro or similar using `/usr/lib64` for 64-bit libraries (versus `/usr/lib` on Debian-based distros).

**Issue description:**
Our current logic to copy shared mono libraries to the export templates' `data.mono.*` folders rely on `mono_root + 'lib'`:
https://github.com/godotengine/godot/blob/5a23ab61fae79cce576fe7ec60fd928d2bd27fca/modules/mono/build_scripts/mono_configure.py#L389-L396

This fails on 64-bit RPM-based distros where mono is installed in `/usr/lib64` instead of `/usr/lib`.

We should rely on `pkg-config` when available to figure out what is the proper lib folder, and otherwise the logic should be (pseudo code):
```
if 64bit build:
  if lib64 exists: use lib64 (rpm)
  else: use lib (deb)
elif 32bit build:
  if lib32 exists: use lib32 (deb)
  else: use lib (rpm)
```

That's not all though, as mono hardcodes those installation paths, so trying to load a fix `data.mono` with the libraries in `Mono/lib` will still fail:
```
System.TypeInitializationException: The type initializer for 'System.Random' threw an exception. ---> System.DllNotFoundException: /home/akien/tmp/godot/test-export/lin/data_DodgeTheCreepsCS/Mono/lib/../lib64/libmono-native.so
```

So the data folder would need to have:
```
Mono/bin/*
Mono/lib/  (empty)
Mono/lib64/*
```

I'm not sure how we can influence the base path before `../lib64/libmono-native.so`, but if we can change it, it should likely point to `Mono/bin/` instead of `Mono/lib/`, so that we can have either:
```
Mono/bin/*
Mono/lib/*
```
or
```
Mono/bin/*
Mono/lib64/*
```
depending on whether the template was packaged on Ubuntu or CentOS, basically.

**Comment by neikeq** [comment]
>That's not all though, as mono hardcodes those installation paths, so trying to load a fix `data.mono` with the libraries in `Mono/lib` will still fail:
>```
>System.TypeInitializationException: The type initializer for 'System.Random' threw an exception. ---> >System.DllNotFoundException: /home/akien/tmp/godot/test-export/lin/data_DodgeTheCreepsCS/Mono/lib/../lib64/libmono-native.so
>```

We can use a dlopen fallback to find the so at the right location, just like we're doing on Android.

**Comment by KoBeWi** [comment]
Can anyone still reproduce this bug in Godot 3.2.3 or any later release?

**Comment by akien-mga** [comment]
Yes, this is still reproducible any time I do a local build on my system (Mageia).

With this hack + manually creating a `lib64` symlink to `lib` in `bin/GodotSharp/Mono/` I can then use the compiled binary properly:
```diff
diff --git a/modules/mono/build_scripts/mono_configure.py b/modules/mono/build_scripts/mono_configure.py
index 6869627873..ec731a4773 100644
--- a/modules/mono/build_scripts/mono_configure.py
+++ b/modules/mono/build_scripts/mono_configure.py
@@ -510,6 +510,8 @@ def copy_mono_shared_libs(env, mono_root, target_mono_root_dir):
     def copy_if_exists(src, dst):
         if os.path.isfile(src):
             copy(src, dst)
+            return True
+        return False
 
     platform = env["platform"]
 
@@ -562,7 +564,8 @@ def copy_mono_shared_libs(env, mono_root, target_mono_root_dir):
             ]
 
         for lib_file_name in lib_file_names:
-            copy_if_exists(os.path.join(mono_root, "lib", lib_file_name), target_mono_lib_dir)
+            if not copy_if_exists(os.path.join(mono_root, "lib64", lib_file_name), target_mono_lib_dir):
+                copy_if_exists(os.path.join(mono_root, "lib", lib_file_name), target_mono_lib_dir)
 
 
 def pkgconfig_try_find_mono_root(mono_lib_names, sharedlib_ext):
```

**Comment by rhofour** [comment]
Any reason that hack isn't good enough to merge?

I've been failing to get a locally built binary to work on Fedora and it seems like this is why.

**Comment by akien-mga** [comment]
> Any reason that hack isn't good enough to merge?

It only does part of what's needed, I still need to copy or symlink `lib64` to `lib` manually, so it's not a good solution. The symlink could also be created from Python but that too is too much of a hack.

Might be possible to copy `/usr/lib64` and rename to `lib` though.
